tools/ansible/tasks/wsu/main.yaml

```diff
@@ -163,19 +163,30 @@
       retries: 12
       delay: 5
 
-    - name: Get hybrid overlay subnet
-      delegate_to: localhost
-      shell: oc get nodes "{{ node_name.stdout }}" -o=jsonpath='{.metadata.annotations.k8s\.ovn\.org/hybrid-overlay-hostsubnet}'
-      register: hybrid_subnet
-
     - name: Create cni config directory
       win_file:
         path: "{{ win_temp_dir.path }}\\cni\\config"
         state: directory
 
-    - name: Create cni config
-      win_copy:
-        content: "{\"cniVersion\":\"0.2.0\",\"name\":\"OpenShiftNetwork\",\"type\":\"win-overlay\",\"capabilities\":{\"dns\":true},\"ipam\":{\"type\":\"host-local\",\"subnet\":\"{{ hybrid_subnet.stdout }}\"},\"policies\":[{\"name\":\"EndpointPolicy\",\"value\":{\"Type\":\"OutBoundNAT\",\"ExceptionList\":[\"10.132.0.0/16\"]}},{\"name\":\"EndpointPolicy\",\"value\":{\"Type\":\"ROUTE\",\"DestinationPrefix\":\"10.132.0.0/16\",\"NeedEncap\":true}}]}"
+    # Get the subnet associated with host. We assume the network operator object has been modified to
+    # include hybrid overlay config
+    - name: Get the subnet associated with host
+      delegate_to: localhost
+      shell: "oc get nodes {{ node_name.stdout}} -o=jsonpath='{.metadata.annotations.k8s\\.ovn\\.org\\/hybrid-overlay-hostsubnet}'"
+      register: ovn_host_subnet
+
+    # Get the service CIDR associated with the OpenShift network operator object. We assume that network always object
+    # has atleast one entry as per defaulting at
+    # https://github.com/openshift/installer/blob/master/pkg/types/defaults/installconfig.go#L18
+    - name: Get the service CIDR associated with network object
+      delegate_to: localhost
+      shell: "oc get network.operator.openshift.io/cluster -o=jsonpath='{.spec.serviceNetwork[0]}'"
```

Comment:
Do we always want to use the first element in the array? What are the other elements?

Reply:
We can have multiple CIDRs, as of now it is limited to 1. Have a look at https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_network.crd.yaml#L136

Reply:
I think so at least for now. There could be a more complex network setup that has multiple service networks but lets cross that bridge later. Make comment about that here.